x/btcstaking/keeper/power_dist_change.go

```diff
@@ -238,6 +261,15 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 		// if this finality provider is slashed, continue to avoid
 		// assigning delegation to it
 		if _, ok := slashedFPs[fpBTCPKHex]; ok {
+			fp.IsSlashed = true
+
+			statusChangeEvent := types.NewFinalityProviderStatusChangeEvent(fp.BtcPk, types.FinalityProviderStatus_STATUS_SLASHED)
+			if err := sdkCtx.EventManager().EmitTypedEvent(statusChangeEvent); err != nil {
+				panic(fmt.Errorf(
+					"failed to emit FinalityProviderStatusChangeEvent with status %s: %w",
+					types.FinalityProviderStatus_STATUS_SLASHED.String(), err))
+			}
```

Comment:
Can this be done in the message handler of finality signatures? Doing it here might add more repsonsibilities to this function, which was supposed to focus on computing the voting power table

Reply:
We already have delegation status change event emitted here https://github.com/babylonlabs-io/babylon/blob/c33f66f0a235c759db907ac16f8f8cc1466129cb/x/btcstaking/keeper/power_dist_change.go#L213. I thought it would be good to put these types of events in one place

Reply:
I think I agree with this i.e in case of `FinalityProviderStatusChangeEvent` the way to go is to keep it as on type of even i.e if we can generate all state transitions in transaction context we should do it. If we can't , then it must be a block event.  

Having that in mind, imo we need to generate all `FinalityProviderStatusChangeEvent` in block context as some state transitions do not happen due to transactions like for example, becoming jailed.

Reply:
Yep, imo `slashing` and status change are different types of events that may interest to different types of subscribers. For subscribers (this PR targets on) that are interested in status change, they should be able to get all the status change event within `block_events`